common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/BlockTransferMessage.java

```diff
@@ -42,7 +43,7 @@
   /** Preceding every serialized message is its type, which allows us to deserialize it. */
   public enum Type {
     OPEN_BLOCKS(0), UPLOAD_BLOCK(1), REGISTER_EXECUTOR(2), STREAM_HANDLE(3), REGISTER_DRIVER(4),
-    HEARTBEAT(5);
+    HEARTBEAT(5), REGISTER_KUBERNETES_APP(6);
```

Comment:
But the alternative for introducing cluster manager specific ideas into the code base is going to be more troublesome IMO (and harder time to merge). I don't believe we added it just for Mesos, although probably only the Mesos integration end up using it.